runtime/compiler/control/CompilationThread.cpp

```diff
@@ -8101,6 +8101,110 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
    return startPC;
    }
 
+bool
+TR::CompilationInfoPerThreadBase::isCompilingWithExcessiveCacheSize(TR::CompilationInfoPerThreadBase *compilationInfo, CompileParameters *compileParameters)
+   {
+      J9JITConfig *jitConfig = compilationInfo->_jitConfig;
+      J9VMThread *vmThread = compileParameters->_vmThread;
+
+      if (!(jitConfig->runtimeFlags & (J9JIT_CODE_CACHE_FULL | J9JIT_DATA_CACHE_FULL)))
+         {
+            return false;
+         }
+
+      // Optimization to disable future first time compilations from reaching the queue
+      compilationInfo->getCompilationInfo()->getPersistentInfo()->setDisableFurtherCompilation(true);
+
+      if (TR::Options::isAnyVerboseOptionSet(TR_VerboseCompileEnd, TR_VerboseCompFailure))
+         {
+         TR_VerboseLog::writeLineLocked(TR_Vlog_INFO,"<WARNING: JIT CACHES FULL> Disable further compilation");
+         }
+      if (jitConfig->runtimeFlags & J9JIT_CODE_CACHE_FULL)
+         Trc_JIT_cacheFull(vmThread);
+      if (jitConfig->runtimeFlags & J9JIT_DATA_CACHE_FULL)
+         Trc_JIT_dataCacheFull(vmThread);
+      compilationInfo->_methodBeingCompiled->_compErrCode = compilationExcessiveSize;
+
+      return true;
+   }
+
+bool
+TR::CompilationInfoPerThreadBase::isCompilingRestrictedMethod(TR::CompilationInfoPerThreadBase *compilationInfo, TR_ResolvedMethod *compilee, CompileParameters *compileParameters, TR_FilterBST *&filterInfo)
+   {
+      TR_J9VMBase *vm = compileParameters->_vm;
+      J9VMThread *vmThread = compileParameters->_vmThread;
+
+      // JITServer: methodCanBeCompiled check should have been done on the client, skip it on the server.
+      if (compilationInfo->_methodBeingCompiled->isOutOfProcessCompReq() || compilationInfo->methodCanBeCompiled(compileParameters->trMemory(), vm, compilee, filterInfo))
+         {
+            return false;
+         }
+
+      compilationInfo->_methodBeingCompiled->_compErrCode = compilationRestrictedMethod;
+
+      TR::Options *options = TR::Options::getJITCmdLineOptions();
+      if (vm->isAOT_DEPRECATED_DO_NOT_USE())
+         options = TR::Options::getAOTCmdLineOptions();
+      if (options->getVerboseOption(TR_VerboseCompileExclude))
+         {
+         TR_VerboseLog::writeLineLocked(TR_Vlog_COMPFAIL, "%s cannot be translated", compilee->signature(compileParameters->trMemory()));
+         }
+      Trc_JIT_noAttemptToJit(vmThread, compilee->signature(compileParameters->trMemory()));
+
+      return true;
+   }
+
+void
+TR::CompilationInfoPerThreadBase::setCompiledMethodOptimzationPlan(TR::CompilationInfoPerThreadBase *compilationInfo, TR_ResolvedMethod *compilee, TR_J9VMBase *vm)
```

Comment:
I think this is incorrectly named, it should be something along the lines of `addUpgradeHintInSCCIfNeeded`.